analysis_for_run.py

```diff
@@ -0,0 +1,236 @@
+    tmp.pop(0)
+    tmp.pop(1)
+    tmp.pop(-1)
+    tmp.pop(-1)
+    tmp=[x.replace('|', '') for x in tmp]
+    tmp=[re.sub(r"\s+", ' ', x) for x in tmp]
+    return tmp
+
+# Function to handle the stdout of the ICA CLI tool. Returns a list of list of strings.
+def icav_out(bashCommand):
+    process = subprocess.Popen(bashCommand.split(), stdout=subprocess.PIPE)
+    output, error = process.communicate()
+    tmp=output.decode("utf-8")
+    tmp=tmp.split('\n')
+    tmp.pop(0)
+    tmp.pop(-1)
+    tmp.pop(-1)
+    tmp=[x.replace('\t', ' ') for x in tmp]
+    return tmp
+
+# Function to loop through files needed to be downloaded for each analysis.
+def ica_download(target_dir,files_list):
+    for i in files_list:
+        bashCommand = "icav2 projectdata download %s%s /Volumes/IDGenomics_NAS/WGS_Serotyping/%s/" % (target_dir,i,args.run_name)
+        process = subprocess.Popen(bashCommand.split(), stdout=subprocess.PIPE)
+        process.communicate()
+
+# Slack_sdk values needed for sending messages to Slack. Uses an API already set up on the Slack website for the UPHL Workspace to post messages on the notifications channel.
+# Function makes sending Slack messages as easy as using the print funcition.
+client = WebClient(token=config.token)
+channel_id = config.channel_id
+def slack_message(string):
+    try:
+
+        result = client.chat_postMessage(
+            channel=channel_id,
+            text=string
+        )
+
+    except:
+        print("Slack Error")
+
+slack_message('Monitoring %s on BSSH' % args.run_name)
+print('Monitoring %s on BSSH' % args.run_name)
+# This While loops Uses the BaseSpace CLI tool to monitor the progress of the run
+t=0
+while t==0:
+    for i in configurations:
+        bashCommand='bs list --config=%s runs' % i
+        tmp=bs_out(bashCommand)
+        tmp=pd.DataFrame(index=[row.split()[2] for row in tmp[1:]], columns=tmp[0].split()[1:], data=[row.split()[1:4] for row in tmp[1:]])
+        try:
+            if tmp.at[args.run_name,'Status']=='Complete':
+                user=i
+                idd=tmp.at[args.run_name,'Id']
+                slack_message('%s is "Complete" on BSSH' % args.run_name)
+                print('%s is "Complete" on BSSH' % args.run_name)
+                t=1
+                break
+        except:
+            continue
+        try:
+            if tmp.at[args.run_name,'Status']=='Failed' or tmp.at[args.run_name,'Status']=='Stopped' or tmp.at[args.run_name,'Status']=='Needs Attention' or tmp.at[args.run_name,'Status']=='Timed Out':
+                user=i
+                idd=tmp.at[args.run_name,'Id']
+                slack_message('%s has "Failed" or was unable to complete on BSSH; Script is aborting, check BSSH for more information' % args.run_name)
+                print('%s has "Failed" or was unable to complete on BSSH; Script is aborting, check BSSH for more information' % args.run_name)
+                sys.exit()
+        except:
+            continue
```

Comment:
I feel like these can likely be combined into one if statement

```python
if tmp.at[args.run_name,'Status']=='Complete':
<stuff>
elif tmp.at[args.run_name,'Status']=='Failed' or tmp.at[args.run_name,'Status']=='Stopped' or tmp.at[args.run_name,'Status']=='Needs Attention' or tmp.at[args.run_name,'Status']=='Timed Out':
<different stuff>
else:
<other stuff>
```

Reply:
This is a bit goofy but just if statements won't work with the function I am using, pandas.df.at, which errors out if the dataframe is empty which it will be often be.